### ISSUE TITLE
investigate feasibility of swagger

### ISSUE THREAD

**Issue by davidwallis3101** [Bounz/HomeGenie-BE#76]
investigate adding swagger to document the api

**Comment by davidwallis3101** [comment]
postman might be more practical I think :)

**Comment by Bounz** [comment]
Well, the main goal of swagger (from my point of view) is to

1. document the API
2. provide some kind of contract for 3rd party developers

So swagger would be suitable with HG to replace (or extend) existing documentation of Web API (like https://bounz.github.io/HomeGenie-BE/api/mig/core_api_config.html#2)

**Comment by davidwallis3101** [comment]
Yeah I'm just not sure that its going to work with websocketsharp and more suited to WCF / Rest style services?

**Comment by Bounz** [comment]
Well, I think it's possible, but one of the advantages of Swagger is autogeneration of swagger file and I think it's not possible in the current implementation of the backend.
So I think we should return to this question after doing some moves towards asp.net core.
